library.sh

```diff
@@ -794,11 +794,17 @@ function shellcheck_new_files() {
 }
 
 function latest_version() {
-  local semver=$(git describe --match "v[0-9]*" --abbrev=0)
-  local major_minor=$(echo "$semver" | cut -d. -f1-2)
+  if [ $(current_branch) = "master" ] || [ $(current_branch) = "main" ]; then
```

Comment:
I would rather not hardcode these in, they are user set-able and programmatically discovery-able.

Reply:
I didn't find a reliable way to know the default branch. How about just checking `! is_release`? I tend to minimize the impact, so specifying the branch names means less surprise to me, but I am OK with that too.

Reply:
`! is_release_branch` should be okay

Reply:
(or better reverse the if/else, so `if is_release_branch; then`)

Reply:
This might be a little bit tricky when running using a local branch that is targeting to be merged to a release branch. e.g. if you are working on a local branch based on v0.18, this latest_version will show v0.18 before, but now since the name of local branch is not a release branch, it will not use `git descirbe` so it will find 0.20 instead.

Reply:
In this PR you are pushing from `zhongduo:fix-latest-version` to `knative:master`. This will be broken on your local copy, won't it?

Reply:
I am not sure what you mean by "broken" here. But the problem in my previous comment will only happen when I am basing on a release branch if I change this PR to use `is_release_branch`.

Reply:
You don't want to use `is_release_branch` because is checks the branch name against a regex to see if it a release. This is valid point. But the current encoded logic is also to check against the branch name. Isn't this wrong for the same exact reason?

Reply:
Two reasons: 
1. minimal impact: `! is_release_branch` modifies the logic for all branches other than the ~20 release branches, while this PR only modify one branch.
2. For user forked branches with proper tag, using this PR it will continue working in this case. But with `! is_release_branch`, all the user local branches targeting release branches will fail.

Reply:
This solution won't work on forks of eventing-kafka master unless people's forked branch is named `master` or `main` until the next successful release.

Reply:
That depends on how you run it. If it is a PR run by Prow it will work as Prow will test master with the PR merged. The only time it will fail is when you run the test locally. AFAIK, most repos have the tagging working properly, so I think the number of failure cases is reasonable.

Reply:
Then IMO trading one set of failures for a different set of failures + unknown possibly complications isn't worth just adding a single tag to eventing-kafka repo and documenting what to do for future release failures.

Reply:
hmm, I am not sure if I will like having both v0.20 and v0.20.0 tags... But I will leave that to the community to make the call.